### DISCUSSION TITLE
changeSelect2 is not fired

### DISCUSSION THREAD

**Discussion by TechAspirant** [serenity-is/Serenity#5774] [Help]
My serene application is in .Net 5. In my case changeSelect2 is not fired when I override PrepareQuery whereas It gets fired when I use 'base.PrepareQuery(query)' in PrepareQuery.
I want to make distinct lookup editor and for that I created my custom lookup editor with overriding  PrepareQuery.

Below code when changeSelect2 not fired
```
protected override void PrepareQuery(SqlQuery query)
        {
            var fld = Entities.PayloadDefinitionRow.Fields;
            query.Distinct(true)
                .Select(fld.ManufacturerManufacturerName);
            
            AddTenantFilter(query);
        }
```

Code when changeSelect2 gets fired
```
protected override void PrepareQuery(SqlQuery query)
        {
            base.PrepareQuery(query);
            AddTenantFilter(query);
        }
```

**Comment by eissa1005** [comment]
@TechAspirant 
Try Use Criteria
 protected override void PrepareQuery(SqlQuery query)
        {
            var fld = Entities.CustomerRow.Fields;
            query.Distinct(true)
                .Select(fld.Country)
                .Select(fld.City)
                .Where(
                    new Criteria(fld.Country) != "" &
                    new Criteria(fld.Country).IsNotNull() &
                    new Criteria(fld.City) != "" &
                    new Criteria(fld.City).IsNotNull());
        }
